scripts/getReleaseStatus.js

```diff
@@ -27,33 +28,67 @@ async function cacheResponses(body) {
       throw new Error(err)
     })
     .connect()
+}
 
+async function cacheResponses(body) {
+  const client = await getRedisClient()
   const resp = await client.set('applicationInfo', JSON.stringify(body))
   await client.disconnect()
 
   console.log(`Successfully cached application info`, body)
   return resp
 }
 
+async function getStoredData() {
+  const client = await getRedisClient()
+  const responseString = await client.get('applicationInfo')
+  await client.disconnect()
+
+  return JSON.parse(responseString)
+}
+
 const getData = async () => {
-  const responses = await Promise.all(
-    endpoints.map(app => getApplicationInfo(app.infoUrl[process.env.ENVIRONMENT_NAME])),
+  const storedData = await getStoredData()
+
+  const responses = await Promise.allSettled(
+    endpoints.map(app =>
+      getApplicationInfo(app.urlEnv ? `${process.env[app.urlEnv]}/info` : app.infoUrl[process.env.ENVIRONMENT_NAME]),
+    ),
   )
 
-  const body = responses
+  const newData = responses
     .map(response => {
+      if (response.status !== 'fulfilled') {
+        console.log(`Failed to get application info`, response.reason)
+        return undefined
+      }
+      const { body, request } = response.value
       const applicationName = endpoints.find(
-        app => response.request.url === app.infoUrl[process.env.ENVIRONMENT_NAME],
+        app =>
+          request?.url === (app.urlEnv ? `${process.env[app.urlEnv]}/info` : app.infoUrl[process.env.ENVIRONMENT_NAME]),
       )?.application
       if (!applicationName) return undefined
 
       return {
         app: applicationName,
-        activeAgencies: response.body.activeAgencies,
+        activeAgencies: body.activeAgencies,
       }
     })
     .filter(Boolean)
 
+  // use new data if we have it for app, use stored data if we don't have it.
+  // append any new apps not in stored data
+  // if we have no stored data use new data for all
+  const body = storedData
+    ? storedData
+        .map(stored => {
+          const newApp = newData.find(newApp => newApp.app === stored.app)
+          if (!newApp) return stored
+          return newApp
+        })
+        .concat(newData.filter(newApp => !storedData.find(stored => stored.app === newApp.app)))
```

Comment:
We've only got 2 apps atm, but once we have more not sure this `find` filter is what you need, as I think you would be able to have multiple apps that need filtering out?

Reply:
Good spot, will update